src/connector/src/sink/big_query.rs

```diff
@@ -68,27 +88,44 @@ fn default_max_batch_rows() -> usize {
 }
 
 impl BigQueryCommon {
-    pub(crate) async fn build_client(&self, aws_auth_props: &AwsAuthProps) -> Result<Client> {
-        let service_account = if let Some(local_path) = &self.local_path {
-            let auth_json = std::fs::read_to_string(local_path)
-                .map_err(|err| SinkError::BigQuery(anyhow::anyhow!(err)))?;
-            serde_json::from_str::<ServiceAccountKey>(&auth_json)
-                .map_err(|err| SinkError::BigQuery(anyhow::anyhow!(err)))?
+    async fn build_client(&self, aws_auth_props: &AwsAuthProps) -> Result<Client> {
+        let auth_json = self.get_auth_json_from_path(aws_auth_props).await?;
+
+        let service_account = serde_json::from_str::<ServiceAccountKey>(&auth_json)
+            .map_err(|err| SinkError::BigQuery(anyhow::anyhow!(err)))?;
+        let client: Client = Client::from_service_account_key(service_account, false)
+            .await
+            .map_err(|err| SinkError::BigQuery(anyhow::anyhow!(err)))?;
+        Ok(client)
+    }
+
+    async fn build_writer_client(
+        &self,
+        aws_auth_props: &AwsAuthProps,
+    ) -> Result<StorageWriterClient> {
+        let auth_json = self.get_auth_json_from_path(aws_auth_props).await?;
+
+        let credentials_file = CredentialsFile::new_from_str(&auth_json)
+            .await
+            .map_err(|e| SinkError::BigQuery(e.into()))?;
+        let client = StorageWriterClient::new(credentials_file).await?;
+        Ok(client)
+    }
+
+    async fn get_auth_json_from_path(&self, aws_auth_props: &AwsAuthProps) -> Result<String> {
+        if let Some(local_path) = &self.local_path {
```

Comment:
Maybe not related to this PR: can we possibly not use separate `local_path` and `s3_path` and use a single option `path`? The path type can be distinguished by the path prefix such as `file://` and `s3://`? 

Besides, bigquery is on google cloud platform while users should upload the auth file to S3. It looks strange to me. Does gcp have S3 compatible object store?

Reply:
agreed, references to s3 or aws in general always confused me. Is there agcp alternative?

Reply:
Unfortunately, this file is necessary to connect to gcp, so it can't be saved at gcp, another alternative is to import the json from this file by means of the string

Reply:
May provide an extra option for users to include the raw auth string in the options.

Can do it in a separate PR since the current PR is already large and this feature is independent to the current PR.